tests/backends/base/test_creation.py

```diff
@@ -40,3 +45,49 @@ def test_custom_test_name_with_test_prefix(self):
         test_connection.settings_dict['TEST'] = {'NAME': test_name}
         signature = BaseDatabaseCreation(test_connection).test_db_signature()
         self.assertEqual(signature[3], test_name)
+
+
+class CreateTestDbTests(TransactionTestCase):
+    available_apps = []
+
+    @mock.patch.object(MigrateCommand, 'handle', return_value='')
+    def test_migrate_test_setting_false(self, migrate_mock):
+        old_database_name = settings.DATABASES['default']['NAME']
+        test_settings = copy.deepcopy(settings.DATABASES['default'])
+        test_settings['TEST']['NAME'] = '%s%s' % (type(self).__name__, os.getpid())
+        test_settings['TEST']['MIGRATE'] = False
+        connections = ConnectionHandler({'default': test_settings})
+        creation = connections['default'].creation
+        creation.create_test_db(verbosity=0, autoclobber=True)
+        try:
+            migrate_mock.assert_not_called()
+        finally:
+            creation.destroy_test_db(old_database_name, verbosity=0)
+
+    @mock.patch.object(MigrateCommand, 'handle', return_value='')
+    def test_migrate_test_setting_true(self, migrate_mock):
+        old_database_name = settings.DATABASES['default']['NAME']
+        test_settings = copy.deepcopy(settings.DATABASES['default'])
+        test_settings['TEST']['NAME'] = '%s%s' % (type(self).__name__, os.getpid())
+        test_settings['TEST']['MIGRATE'] = True
+        connections = ConnectionHandler({'default': test_settings})
+        creation = connections['default'].creation
+        creation.create_test_db(verbosity=0, autoclobber=True)
+        try:
+            migrate_mock.assert_called_once()
+        finally:
+            creation.destroy_test_db(old_database_name, verbosity=0)
+
+    @mock.patch.object(MigrateCommand, 'handle', return_value='')
+    def test_migrate_test_setting_unset(self, migrate_mock):
+        old_database_name = settings.DATABASES['default']['NAME']
+        test_settings = copy.deepcopy(settings.DATABASES['default'])
+        test_settings['TEST']['NAME'] = '%s%s' % (type(self).__name__, os.getpid())
+        del test_settings['TEST']['MIGRATE']
+        connections = ConnectionHandler({'default': test_settings})
+        creation = connections['default'].creation
+        creation.create_test_db(verbosity=0, autoclobber=True)
+        try:
+            migrate_mock.assert_called_once()
+        finally:
+            creation.destroy_test_db(old_database_name, verbosity=0)
```

Comment:
New tests are failing on Oracle. I think we should avoid creating a test database, maybe by mocking `_create_test_db()` :thinking:

Reply:
I'm not sure that will work as the final line in `create_test_db()` is a call to `.ensure_connection()`. I don't have Oracle setup to test, but trying the technique on Postgresql results in:

```
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/mock.py", line 1342, in patched
    return func(*newargs, **newkeywargs)
  File "/home/jon/devel/django/tests/backends/base/test_creation.py", line 71, in test_migrate_test_setting_false
    creation.create_test_db(verbosity=0, autoclobber=True, serialize=False)
  File "/home/jon/devel/django/django/db/backends/base/creation.py", line 86, in create_test_db
    self.connection.ensure_connection()
...
django.db.utils.OperationalError: FATAL:  database "CreateTestDbTests112547" does not exist
```

I'm open to other suggestions, but would it be reasonable to disable this test for Oracle?

Here is my WIP patch if you want to play around with it:

```diff
diff --git a/tests/backends/base/test_creation.py b/tests/backends/base/test_creation.py
index 001a807777..4b4f0e81ad 100644
--- a/tests/backends/base/test_creation.py
+++ b/tests/backends/base/test_creation.py
@@ -8,6 +8,9 @@ from django.db import DEFAULT_DB_ALIAS, connections
 from django.db.backends.base.creation import (
     TEST_DATABASE_PREFIX, BaseDatabaseCreation,
 )
+from django.db.backends.postgresql.creation import (
+    DatabaseCreation as OracleDatabaseCreation,
+)
 from django.db.utils import ConnectionHandler
 from django.test import SimpleTestCase, TransactionTestCase
 
@@ -47,46 +50,51 @@ class TestDbSignatureTests(SimpleTestCase):
         self.assertEqual(signature[3], test_name)
 
 
+@mock.patch.object(OracleDatabaseCreation, '_destroy_test_db')
+@mock.patch.object(OracleDatabaseCreation, '_create_test_db')
+@mock.patch.object(MigrateCommand, 'handle', return_value='')
 class CreateTestDbTests(TransactionTestCase):
     available_apps = []
 
-    @mock.patch.object(MigrateCommand, 'handle', return_value='')
-    def test_migrate_test_setting_false(self, migrate_mock):
+    def test_migrate_test_setting_false(
+        self,
+        migrate_mock,
+        _oracle_create_mock,
+        _oracle_destroy_mock,
+    ):
         old_database_name = settings.DATABASES['default']['NAME']
         test_settings = copy.deepcopy(settings.DATABASES['default'])
         test_settings['TEST']['NAME'] = '%s%s' % (type(self).__name__, os.getpid())
         test_settings['TEST']['MIGRATE'] = False
         connections = ConnectionHandler({'default': test_settings})
         creation = connections['default'].creation
-        creation.create_test_db(verbosity=0, autoclobber=True)
+        creation.create_test_db(verbosity=0, autoclobber=True, serialize=False)
         try:
             migrate_mock.assert_not_called()
         finally:
             creation.destroy_test_db(old_database_name, verbosity=0)
 
-    @mock.patch.object(MigrateCommand, 'handle', return_value='')
-    def test_migrate_test_setting_true(self, migrate_mock):
+    def test_migrate_test_setting_true(self, migrate_mock, _oracle_create_mock, _oracle_destroy_mock):
         old_database_name = settings.DATABASES['default']['NAME']
         test_settings = copy.deepcopy(settings.DATABASES['default'])
         test_settings['TEST']['NAME'] = '%s%s' % (type(self).__name__, os.getpid())
         test_settings['TEST']['MIGRATE'] = True
         connections = ConnectionHandler({'default': test_settings})
         creation = connections['default'].creation
-        creation.create_test_db(verbosity=0, autoclobber=True)
+        creation.create_test_db(verbosity=0, autoclobber=True, serialize=False)
         try:
             migrate_mock.assert_called_once()
         finally:
             creation.destroy_test_db(old_database_name, verbosity=0)
 
-    @mock.patch.object(MigrateCommand, 'handle', return_value='')
-    def test_migrate_test_setting_unset(self, migrate_mock):
+    def test_migrate_test_setting_unset(self, migrate_mock, _oracle_create_mock, _oracle_destroy_mock):
         old_database_name = settings.DATABASES['default']['NAME']
         test_settings = copy.deepcopy(settings.DATABASES['default'])
         test_settings['TEST']['NAME'] = '%s%s' % (type(self).__name__, os.getpid())
         del test_settings['TEST']['MIGRATE']
         connections = ConnectionHandler({'default': test_settings})
         creation = connections['default'].creation
-        creation.create_test_db(verbosity=0, autoclobber=True)
+        creation.create_test_db(verbosity=0, autoclobber=True, serialize=False)
         try:
             migrate_mock.assert_called_once()
         finally:
```